snappy/snap_file.go

```diff
@@ -229,6 +229,14 @@ func (s *SnapFile) Install(inter progress.Meter, flags InstallFlags) (name strin
 	if err := s.m.addSquashfsMount(s.instdir, inhibitHooks, inter); err != nil {
 		return "", err
 	}
+	// if anything goes wrong we ensure we stop
+	defer func() {
+		if err != nil {
+			if e := s.m.removeSquashfsMount(s.instdir, inter); e != nil {
```

Comment:
In this case there is no (good) reason. Often a different var is used when the original err is needed to compose a new message like `return fmt.Errorf("failed to cleanup from error %s but cleanup failed more horrible: %s", err, e)`. TBH I am not very happy (at all) with the state of Install() as it is now, it grew a bit out of control and the defers (while very useful) make the flow hard to read. I hope that with something like https://github.com/ubuntu-core/snappy/pull/347 (or something else that has a similar spirit) this becomes less of an issue.